alacritty/src/renderer/box_drawing.rs

```diff
@@ -0,0 +1,584 @@
+    let mut canvas = Canvas::new(width, height);
+
+    match character {
+        // Horizonatal dashes: '┄', '┅', '┈', '┉', '╌', '╍'.
+        '\u{2504}' | '\u{2505}' | '\u{2508}' | '\u{2509}' | '\u{254c}' | '\u{254d}' => {
+            let (num_gaps, stroke_size) = match character {
+                '\u{2504}' => (2, stroke_size),
+                '\u{2505}' => (2, heavy_stroke_size),
+                '\u{2508}' => (3, stroke_size),
+                '\u{2509}' => (3, heavy_stroke_size),
+                '\u{254c}' => (1, stroke_size),
+                '\u{254d}' => (1, heavy_stroke_size),
+                _ => unreachable!(),
+            };
+
+            let dash_gap_len = max(width / 8, 1);
+            let dash_len = max(width.saturating_sub(dash_gap_len * num_gaps) / (num_gaps + 1), 1);
+            let y = canvas.center_v();
+            for gap in 0..=num_gaps {
+                let x = min(gap * (dash_len + dash_gap_len), width);
+                canvas.draw_h_line(x as f32, y, dash_len as f32, stroke_size);
+            }
+        },
+        // Vertical dashes: '┆', '┇', '┊', '┋', '╎', '╏'.
+        '\u{2506}' | '\u{2507}' | '\u{250a}' | '\u{250b}' | '\u{254e}' | '\u{254f}' => {
+            let (num_gaps, stroke_size) = match character {
+                '\u{2506}' => (2, stroke_size),
+                '\u{2507}' => (2, heavy_stroke_size),
+                '\u{250a}' => (3, stroke_size),
+                '\u{250b}' => (3, heavy_stroke_size),
+                '\u{254e}' => (1, stroke_size),
+                '\u{254f}' => (1, heavy_stroke_size),
+                _ => unreachable!(),
+            };
+
+            let dash_gap_len = max(height / 8, 1);
+            let dash_len = max(height.saturating_sub(dash_gap_len * num_gaps) / (num_gaps + 1), 1);
+            let x = canvas.center_h();
+            for gap in 0..=num_gaps {
+                let y = min(gap * (dash_len + dash_gap_len), height);
+                canvas.draw_v_line(x, y as f32, dash_len as f32, stroke_size);
+            }
+        },
+        // Horizontal lines: '─', '━', '╴', '╶', '╸', '╺'.
+        // Vertical lines: '│', '┃', '╵', '╷', '╹', '╻'.
+        // Light and heavy line box components:
+        // '┌','┍','┎','┏','┐','┑','┒','┓','└','┕','┖','┗','┘','┙','┚','┛',├','┝','┞','┟','┠','┡',
+        // '┢','┣','┤','┥','┦','┧','┨','┩','┪','┫','┬','┭','┮','┯','┰','┱','┲','┳','┴','┵','┶','┷',
+        // '┸','┹','┺','┻','┼','┽','┾','┿','╀','╁','╂','╃','╄','╅','╆','╇','╈','╉','╊','╋'.
+        // Mixed light and heavy lines: '╼', '╽', '╾', '╿'.
+        '\u{2500}'..='\u{2503}' | '\u{250c}'..='\u{254b}' | '\u{2574}'..='\u{257f}' => {
+            // Left horizontal line.
+            let stroke_size_h1 = match character {
+                '\u{2500}' | '\u{2510}' | '\u{2512}' | '\u{2518}' | '\u{251a}' | '\u{2524}'
+                | '\u{2526}' | '\u{2527}' | '\u{2528}' | '\u{252c}' | '\u{252e}' | '\u{2530}'
```

Comment:
Bunch of glyphs here are more than 2 in sequential order, where `..=` might be beneficial. Did you intentionally write out every single one for consistency to make editing in the future easier without having to break your brain about character ranges?

Reply:
Yeah, and it was also a sanity check, since you should have the same amount of elements in all of match arms and matches, so you won't miss anything. And if you look on some of them they have like 2 at max contiguous sequential elements, which is pretty much the same amount of chars as putting `|`.